### PR TITLE
Expose formatted service name as a `const fn`

### PR DESCRIPTION
**Pull request by mkatychev**
## Motivation

It is oftentimes cumbersome to use `tonic::server::NamedService::NAME` when `T` has multiple generics:
```rust
impl<T: ServerTrait> tonic::server::NamedService for ServiceServer<T> {
    const NAME: &'static str = "service_api.Service";
}
```

This is particularly egregious when using the `HealthReporter` methods since `HealthReporter::set_serving` requires a fully qualified `ServiceServer<T>`:

```rust
let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
health_reporter.set_serving::<MyServer<MyInner<MyLongQualifiedType, MyOtherQualifiedType, T, T2>>>().await;
```

## Solution

expose a `pub const fn service_name() -> &'static str` that returns the formatted string.

### CODE REVIEW

**Comment by AmeliasCode** [comment]
+1 to this! It would make it a lot less verbose to get the service name.

File: tonic-build/src/server.rs [hunk 1]
```diff
@@ -375,8 +375,14 @@ fn generate_named(
     service_name: &str,
 ) -> TokenStream {
     let service_name = syn::LitStr::new(service_name, proc_macro2::Span::call_site());
+    let name_doc = generate_doc_comment(" Generated gRPC service name");
 
     quote! {
+        #name_doc
+        pub const fn service_name() -> &'static str {
```
Comment (djc) [inline tonic-build/src/server.rs:382]: I think it would still make sense for this to a `const SERVICE_NAME: &str`?

Comment (mkatychev) [reply tonic-build/src/server.rs:382]: fair point, should the trait const below be `const NAME: &'static str = SERVICE_NAME;` then?

Comment (djc) [reply tonic-build/src/server.rs]: Might be nice, want to make that change?

**Comment by mkatychev** [comment]
I cannot seem to reproduce this diff locally:
https://github.com/hyperium/tonic/actions/runs/9472891207/job/26134316596#step:7:16

**Comment by mkatychev** [comment]
> @mkatychev you just need to run `cargo run --package codegen`

Hi @LucioFranco  I did that with the given version on macOS and I am not able to produce the comment diff in the run log

**Comment by djc** [comment]
> > @mkatychev you just need to run `cargo run --package codegen`
> 
> Hi @LucioFranco I did that with the given version of protoc (25.1) on macOS and I am not able to produce the comment diff in the run log

Run `cargo update` locally before you run the codegen?

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Thanks!

**Pull request merged**